uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/LeaveRequest.php

```diff
@@ -296,9 +296,10 @@ private static function validateTOILToAccruedAmountIsValid($params) {
       }
     }
 
-    if ($totalProjectedToilForPeriod > $absenceType->max_leave_accrual && !$unlimitedAccrual) {
+    $maxLeaveAccrual = $absenceType->max_leave_accrual;
+    if ($totalProjectedToilForPeriod > $maxLeaveAccrual && !$unlimitedAccrual) {
       throw new InvalidLeaveRequestException(
-        'The TOIL amount plus all approved TOIL for current period is greater than the maximum for this Absence Type',
+        'The maximum amount of leave that you can accrue is '. $maxLeaveAccrual .' days. Please modify the dates of this request',
```

Comment:
I quite like using `sprintf` for these kind of things. Even though it's longer I think it's clearer and easier to edit later (for example if adding another variable).

`$msg = 'there are ' . $numCats . ' cats';`
vs
`$msg = sprintf('there are %d cats', $numCats);`

But of course it's up to you.